web3.js/src/vote-program.ts

```diff
@@ -410,4 +411,28 @@ export class VoteProgram {
       data,
     });
   }
+
+  /**
+   * Generate a transaction to withdraw safely from a Vote account.
+   *
+   * This function was created as a safeguard for vote accounts running validators, `safeWithdraw`
+   * checks that the withdraw amount will not exceed the rent-exempt minimum. If you wish to close
+   * the vote account by withdrawing the full amount, call the `withdraw` method directly.
+   */
+  static async safeWithdraw(
+    params: WithdrawFromVoteAccountParams,
+    connection: Connection,
```

Comment:
Passing in the `Connection` creates an unnecessary dependency at this file, which is just layouts otherwise.  Since the runtime will prevent any accounts from being rent-paying, how about this function takes in the current balance of the vote account, and just checks that `params.lamports < balance`?

Reply:
So the problem with this is that you can't only check that `parms.lamports < balance` you need to check that `params.lamports < balance - minimumRentAmount`.  If after you've withdrawn you do not have enough left over for rent then your vote account will essentially be closed.  I lost 3 epochs worth of rewards because of this issue.  That being said, I think its worth the dependency (otherwise users need to be smart enough to pass in the correct adjusted balance every time they call this) but that's just me.

In the essence of at least having something out there I've updated this to your specifications.